### ISSUE TITLE
<Table/> - significant increase in 'load more' time of new rows

### ISSUE THREAD

**Issue by dorduch** [wix/wix-style-react#4541]
when using `<Table>` component with infinite scrolling, comparing version 6 to 7 we saw an increase in the loading time of the next table rows

**Comment by nitayneeman** [comment]
I found the issue has started since v7.9.0.

I've already encountered performance issues regarding HOCs and specifically our EllipsisHOC so I suspect it's related to this PR somehow:
https://github.com/wix/wix-style-react/pull/4340

We'll keep investigating.

@dorduch can you approve that v7.8.0 works OK for you?

**Comment by dorduch** [comment]
@nitayneeman  can confirm

**Comment by nitayneeman** [comment]
Well, it turns out that the performance degradation apparently comes from the usage of the new `EllipsisHoc` within `<Text/>` and `<Badge/>`:
https://github.com/wix/wix-style-react/blob/master/src/Text/Text.js#L105
https://github.com/wix/wix-style-react/blob/master/src/Badge/Badge.js#L21

When I tried to ditch those usages from the lines above, the loading time seems to behave as before.

As described in the official docs, it isn't recommended to create a HOC inside a render function, and they emphasize that, so it might lead to performance issues:
https://reactjs.org/docs/higher-order-components.html#dont-use-hocs-inside-the-render-method

So I tried to change the `EllipsisHoc` to this code:
```jsx
export default React.forwardRef(({ Component, props }, ref) => {
  return <Component ref={ref} {...props} />;
});
```
And the performance improved a little bit, but not as expected. Notice our scenario renders hundred of `<Text/>`s and `<Badge/>`s so it actually makes sense to get degradation.

The suggestion from React is to avoid dynamically initializing of HOCs, by making them static. But the thing in our case is we need them to be dynamic because they depend on the component props - so the only way I can think of is to use the constructor and lifecycle hooks somehow to initialize the instance once. After that I believe we'll still have a degradation but probably better than now.

@mykas what do you think?

**Comment by mykas** [comment]
The EllipsisHOC was based on functions and I used `memo` feature from React that suppose to cache the output when props are equal. It could be that the caching mechanism is not working for some reason.  So I suggest checking there.

**Comment by dorduch** [comment]
Any news? We currently using an old version of wsr because of it

**Comment by nitayneeman** [comment]
@mykas Any chance you can continue researching this issue based on your understanding?

**Comment by mykas** [comment]
Hey @nitayneeman so we talked with Simas about this in EllipsisHOC and he used `useMemo` hook to cache the hoc that was causing this little performance issues. Basically useHook is caching the other hook and only rerenders new how when tooltip props are changing.

**Comment by shlomitc** [comment]
@dorduch can ou try installing this version and see if it works better? `@wix/wix-style-react@0.0.0-72e870b8c3ee04cd3e177c41b5a6cd8b`

**Comment by dorduch** [comment]
works good!

**Comment by shlomitc** [comment]
The fix was released in v7.26.0. thanks to everyone! (cc @simast)
